pkg/network/prober/prober.go

```diff
@@ -168,14 +175,18 @@ func (m *Manager) doAsync(ctx context.Context, target string, arg interface{}, p
 			m.keys.Delete(target)
 		}()
 		var (
-			result bool
-			err    error
+			err   error
+			inErr error
 		)
 
 		err = wait.PollImmediate(period, timeout, func() (bool, error) {
-			result, err = Do(ctx, m.transport, target, ops...)
-			return result, err
+			inErr = Do(ctx, m.transport, target, ops...)
+			if inErr != nil {
+				// Do not return error when getting verifierError as retry is expected until timeout.
+				return false, nil
+			}
+			return true, nil
 		})
-		m.cb(arg, result, err)
+		m.cb(arg, inErr == nil, err)
```

Comment:
I don't think this will return the inner error as you wanted to. Can we add a test to the prober that ensures that the inner error and not the timeout error is returned in the end?

Reply:
According to the current test, the timeout error is expected and it is returned actually.

https://github.com/knative/serving/blob/e8fed59edb6ed2f3963e75290b9de271a6b3981e/pkg/network/prober/prober_test.go#L215

I added log message for debug to output inner error from Do().